workers/unity/Assets/Gdk/TestUtils/ComponentObjectSetter.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Reflection;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.TestUtils
+{
+    public static class ComponentObjectSetter
+    {
+        // Reflection magic to get the internal method "SetComponentObject" on the specific EntityManager instance. This is required to add Components to Entities at runtime
+        private static readonly MethodInfo setComponentObjectMethodInfo =
+            typeof(EntityManager).GetMethod("SetComponentObject", BindingFlags.Instance | BindingFlags.NonPublic, null,
+                new Type[] { typeof(Entity), typeof(ComponentType), typeof(object) },
+                new ParameterModifier[] { });
+
+        public static void AddAndSetComponentObject<T>(Entity entity, T component, EntityManager entityManager)
+            where T : Component
+        {
+            if (!entityManager.HasComponent<T>(entity))
+            {
+                entityManager.AddComponent(entity, typeof(T));
+            }
+
+            setComponentObjectMethodInfo.Invoke(entityManager,
```

Comment:
Revert this change. Calling `Invoke` is magnitudes slower than binding to an action and invoking the action.

Reply:
See: http://blogs.msmvps.com/jonskeet/2008/08/09/making-reflection-fly-and-exploring-delegates/

Reply:
This is a helper that's supposed to be only used for unit tests. My thought process here was that I'd rather have less boilerplate code for that case.

Reply:
Ah right, I missed that :)

Reply:
Why not store it in an action anyway? It would be easier to read at the end what exactly is being called.

Reply:
Remove the class altogether as the test case requiring the help er has become obsolete.